---
title: "About Me"
author: "Lily Durkee"
output:
  html_document:
    toc: yes
  word_document:
    toc: yes
  pdf_document:
    toc: yes
bibliography: references.bib
---

# Who I am and where I came from

I grew up in the Washington, DC area in a Maryland suburb called Takoma Park. I always loved the outdoors and insects in particular. I remember when the Brood X cicadas emerged in 2004. Every day at recess, I would hang out with the cicadas. I thought they were the cutest, sweetest things, especially the way they wrapped their legs around my fingers. 

My first exposure to coding occurred in middle school computer science class. We first learned to code in `True BASIC`. While not the most applicable programming language, I am grateful to have learned the basics of if-else statements, for loops, and do loops. In high school, we mostly coded in `Java`. 

I found coding in high school to be frustrating and intimidating. In college, I decided to major in Biology to move away from math and computer science at the University of Maryland, College Park. However, I soon found myself falling in love with insect ecology, and I started learning how to use `R` for data analysis (coding + statistics - the two areas I thought I was avoiding!). I quickly realized that my background in object-based coding was a huge help, and the rigorous math curriculum from my high school days really helped me understand the probability & statistics needed to analyze data effectively. 

After 22 years living in Maryland, I decided I wanted to go to graduate school somewhere with bigger mountains, and I wanted to pursue a higher degree in ecology. The [Graduate Degree Program in Ecology](ecology.colostate.edu) here at CSU seemed to be the perfect fit. I found a home in the [Hufbauer Lab](hufbauerlab.org) and have spent 4.5 years in Fort Collins as I pursue my PhD. I love the proximity to the Rockies, the ample trails along the foothills, and I spend most evenings in the summer paddling down the Cache la Poudre River. The numerous local breweries and coffee shops is a plus, too! I absolutely love it here. After graduation, I hope to stay in the area and work as a federal scientist at USGS or the USDA. 

My top four leisure (or not so leisurely) activities are:

1. Whitewater kayaking - from after work runs to multi-day river trips!
2. Enjoying good food & beer with friends
3. Snow skiing, both in resorts and in the backcountry
4. Mountain biking, particularly in the Utah desert


![A photo of me catching butterflies on Rabbit Ears Pass in 2022.](/Users/lilydurkee/Desktop/Field-Lily.jpeg)


# Research Interests

I study evolutionary ecology in insect systems. I have done work in experimental evolution using  *Tribolium castaneum* (red flour beetle) as a model system to study the effects of immigration and admixture on adaptation to novel environments. I am currently using genomic analyses to identify patterns of local adaptation to elevation in the clouded sulfur butterfly (*Colias philodice eriphyle*).

## Influential papers

Two papers that I love are two review papers, one on evolutionary rescue [@carlson2014evolutionary] and one on genetic rescue [@whiteley2015genetic]. I read these papers in my first few weeks of graduate school, and I still go back to them today when I need a precise definition or good example of either process!


## The mathematics behind my research

In my work with *Tribolium*, I measure growth rate using $\lambda$, where:

$$ \lambda = \frac{N_{t+1}}{N_t}$$

Brett Melbourne at CU Boulder created a model for population dynamics of the *Tribolium* system using a modified, stochastic Ricker model [@melbourne2008extinction]. A simple Ricker model is below:

$$ N_{t+1} = N_te^{r(1-\frac{N_t}{k})} $$

## My computing experience

I have used R extensively for all of my data analysis in graduate school. Recently, I have also started using `bash` to work with genomic data on the alpine cluster.  Here is an example of code I used recently to make a PCA. You can see the graph in the following section.

```
cov<-as.matrix(read.table("colias-full-down4.0x.cov"))
names<-read_delim("colias-full-down4.0x-bam-list.ind", delim = " ", col_names=F)

#eigenvalues
e<-eigen(cov)

#make dataframe
e.df <- as.data.frame(e$vectors)
e.df$ID <- names$X1

e.df$name <- substr(e.df$ID, 1, 2)

e.df <- e.df %>% dplyr::select(c(V1, V2, ID, name))

#upload all info about sites
sites <- read.csv("sites-all1.csv")

pca_4.0x <- merge(e.df, sites, by="name", all=T)
pca_4.0x.data <- merge(pca_4.0x, all.depth, by="ID")
pca_4.0x.data$cov <- ifelse(pca_4.0x.data$coverage>4, 4, pca_4.0x.data$coverage)

pca_4.0x.data$category1 <- ifelse(pca_4.0x.data$name=="Eu", "C. eurytheme", 
                            ifelse(pca_4.0x.data$name=="AL", "C. alexandra", pca_4.0x.data$elevation))

pca_4.0x.data$pair <- ifelse(pca_4.0x.data$name=="WY", "WY", 
                       ifelse(pca_4.0x.data$name=="SC", "E2", pca_4.0x.data$pair))


#simple PCA
plot(e$vectors[,1:2], xlim=c(-.1, .1))

#with ggplot
#library(ggplot2)
plot4.0x <- ggplot(data=filter(pca_4.0x.data, name != "AL" & name != "Eu"), aes(x = V1, y = V2, 
                           shape=category1, 
                           color=pair)) +
  geom_point(size=2)+
  labs(x="PC1", y="PC2") 
  
plot4.0x
```

Here is the bash script that I used to generate the `.cov` file above:

```
set -x

source ~/.bashrc
conda activate pcangsd

module load jdk
module load python

##Traditionally use pcangsd with beagle.gz files
pcangsd -beagle colias-full-down4.0x.minInd90.05maf.baq.HiSeq_filt.beagle.gz -out colias-full-down4.0x -threads 4

```

## What I hope to get out of this class

* Better understand the bioinformatics behind the genomic analyses I am currently working on
* Continue my analysis of genomic data to include analyses of allele frequencies corresponding with different phenotypes, as well as evidence of admixture between populations
* Get more comfortable at using programming languages outside of R including bash and python

# Evaluating some R code

Here is the PCA that I showed code for above:

```{r, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE}
library(tidyverse)

setwd("/Users/lilydurkee/OneDrive - Colostate/Grad School/R-Projects-Grad/Colias")

# individual sample depths
all.depth <- read_delim("covSummary.bedtools.txt", delim = " ")
colnames(all.depth) <- c("full_ID", "ID", "site", "coverage")

all.depth$ID <- str_extract(all.depth$full_ID, "(?<=/).*?(?=_)")

lane <- str_match(all.depth$full_ID, "_(.*?)_(.*?)")
all.depth$lane <- lane[,2]

all.depth$site <- substr(all.depth$ID, 1, 2)

all.depth$bam <- sub(".*/([^_]+_[^_]+_[^_]+)_.*", "\\1", all.depth$full_ID)

# all samples
all.samples <- read.csv("all_samples.csv")

all.depth <- merge(all.depth, all.samples, by="ID", all=T)

# read in covariance matrix
cov<-as.matrix(read.table("colias-full-down4.0x.cov"))
names<-read_delim("colias-full-down4.0x-bam-list.ind", delim = " ", col_names=F)

#eigenvalues
e<-eigen(cov)

#make dataframe
e.df <- as.data.frame(e$vectors)
e.df$ID <- names$X1

e.df$name <- substr(e.df$ID, 1, 2)

e.df <- e.df %>% dplyr::select(c(V1, V2, ID, name))

#upload all info about sites
sites <- read.csv("sites-all1.csv")

pca_4.0x <- merge(e.df, sites, by="name", all=T)
pca_4.0x.data <- merge(pca_4.0x, all.depth, by="ID")
pca_4.0x.data$cov <- ifelse(pca_4.0x.data$coverage>4, 4, pca_4.0x.data$coverage)

pca_4.0x.data$pair <- ifelse(pca_4.0x.data$name=="WY", "WY", 
                       ifelse(pca_4.0x.data$name=="SC", "E2", pca_4.0x.data$pair))


#with ggplot
#library(ggplot2)
plot4.0x <- ggplot(data=filter(pca_4.0x.data, name != "AL" & name != "Eu"), aes(x = V1, y = V2, 
                           shape=elevation, 
                           color=pair)) +
  geom_point(size=2)+
  labs(x="PC1", y="PC2") +
  xlim(-.005, .025) +
  ylim(-.05, .06) 
  
plot4.0x

```

# References

